extensions/ql-vscode/src/cli.ts

```diff
@@ -957,6 +957,9 @@ export class CodeQLCliServer implements Disposable {
   public async getVersion() {
     if (!this._version) {
       this._version = await this.refreshVersion();
+      await commands.executeCommand(
+        'setContext', 'codeql.supportsEvalLog', await this.cliConstraints.supportsPerQueryEvalLog()
+      );
```

Comment:
I like the simplicity of this change. Is this the best place to execute this command, though? It will be invoked every time the extension asks for the CodeQL CLI version, which may be frequent. I expect this command to be executed each time there is a potential change to the configured CLI -- see what the `CliConfigListener` and `QueryServerConfigListener` classes in `config.ts` do, and whether that can be adapted to suit your purpose.

Reply:
This is actually what is happening. The only time `this._version` is undefined is after a config change, where the version may have potentially changed. The version and context are cached until the next time the config changes. By putting it in this if statement, we are guaranteed to call the command the minimal number of times.

Reply:
Fancy. Looks good then! Perhaps add a comment to explain that, so I don't pester you about it again :)